module2/homework/nwdNww/nwdNww.hpp

```diff
@@ -1,11 +1,29 @@
 #pragma once
-
+#include <math.h>
+#include <algorithm>
 int NWD(int lhs, int rhs) {
-    // TODO: Implement me :)
-    return -1;
+    if(lhs==0 || rhs==0){
+        return std::max(lhs, rhs);
+    }
+    lhs = std::abs(lhs);
+    rhs = std::abs(rhs);
+    int temp;
+    while(rhs != 0){
+        temp = rhs;
+        rhs = lhs % rhs;
+        lhs = temp;
+    }
+
+    return lhs;
 }
 
 int NWW(int lhs, int rhs) {
-    // TODO: Implement me :)
-    return -1;
+    if(lhs == 0 || rhs == 0){
+        return 0;
+    }
+    lhs = abs(lhs);
+    rhs = abs(rhs);
+    int result;
+    result = lhs/NWD(lhs, rhs) * rhs;
+    return result;
```

Comment:
I think that creation of variable that is returned 2 lines below is not necessary. 
```suggestion
   return  lhs/NWD(lhs, rhs) * rhs;
```